src/webapp/config.ts

```diff
@@ -0,0 +1,42 @@
+import { mkdtempSync, writeFileSync } from 'fs';
+import { tmpdir } from 'os';
+import { join } from 'path';
+import { PackageLinkConfig } from '.';
+
+interface FrontendPackageLinkConfig {
+  name: string;
+  value: string;
+  displayText?: string;
+}
+
+interface FrontendConfig {
+  packageLinks?: FrontendPackageLinkConfig[];
+}
+
+interface WebappConfigProps {
+  /**
+   * Configuration for custom package page links.
+   */
+  readonly packageLinks?: PackageLinkConfig[];
+}
+
+export class WebappConfig {
+  public readonly path: string;
+  public readonly dir: string;
+  public constructor(private readonly props: WebappConfigProps) {
+    this.dir = mkdtempSync(join(tmpdir(), 'chwebapp'));
+    this.path = join(this.dir, 'config.json');
+    writeFileSync(this.path, JSON.stringify(this.frontendConfig));
+  }
+
+  private get frontendConfig(): FrontendConfig {
+    return {
+      // Remove domains allow list from frontend config
+      packageLinks: (this.props.packageLinks ?? []).map(
+        ({ domains: _, ...rest }) => {
+          return rest;
+        },
+      ),
+    };
```

Comment:
nit: I think this would be a little easier to read if written in a more imperative style - but it might just be personal taste.
```suggestion
    const packageLinks = this.props.packageLinks ?? [];
    const withoutDomains = packageLinks.map(({ domains, ...rest }) => rest);
    return { packageLinks: withoutDomains };
```